### DISCUSSION TITLE
Problem with Typegoose's decorators

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [typegoose/typegoose#856] [Q&A] [answered]
### Screenshots
@typegoose/typegoose's @prop:
![image](https://github.com/typegoose/typegoose/assets/10191085/ebf5a41e-eb55-4533-aa92-cadd5f349c5f)

`
Unable to resolve signature of property decorator when called as an expression.
  Argument of type 'undefined' is not assignable to parameter of type 'Object'.ts(1240)
`

My custom @prop:

![image](https://github.com/typegoose/typegoose/assets/10191085/2fa6a1d1-9130-4a9c-9dfb-9f9ab0fb84da)

### Environment
Node: 18.15.0
VSCode: 1.79.2
typescript: 5.1.3
@typegoose/typegoose: 11.3.0 (Tried many versions down to 10)
mongoose: 7.3.1

**Comment by hasezoey** [comment]
does it work if you use typescript 4.9 or if you use `@prop({})`?

**Reply by angelhdzmultimedia** [reply]
I tried @prop({}) to see if an argument was required, same error.

I'm gonna downgrade typescript to confirm, thank you!

**Comment by hasezoey** [comment, marked as answer]
For future reference, i could reproduce this problem, it is related to not having `experimentalDecorators` set to `true` (meaning using the new way of decorators with [5.0 and forwards](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-5-0.html#decorators)), adding `"experimentalDecorators": true` should fix the issue

~~a future version of typegoose may either support both or switch to the new decorators~~

Edit:
[added a documentation note about this issue](https://typegoose.github.io/typegoose/docs/guides/known-issues#typescript-50-es-decorators)

also as long as the new ES Decorators (stage 3) dont support some kind of metadata support, typegoose will have to rely on `experimentalDecorators` (Stage 2), which will seemingly be possible with typescript 5.2 see [typescript 5.2 iteration plan](https://github.com/microsoft/TypeScript/issues/54298) and [typescript decorator-metadata-proposal implementation PR](https://github.com/microsoft/TypeScript/issues/54657)

**Reply by angelhdzmultimedia** [reply]
Thank you so much. Yes, I set `experimentalDecorators` to `true` as part of my attempts to solve it, 
even thought I assumed I would not need it because official decorators were announced (and because I didn't get an error about having to enable them in tsconfig as usual), but now I realized they haven't been implemented yet. 

### While Typing This Comment...
... i remembered that because I'm working with Nuxt 3, the server directory  NOW requires its own `tsconfig.json`! And that solved it! Because I set `experimentalDecorators: true` in the root `tsconfig.ts` only!!! Hahaha! Thank you!

![image](https://github.com/typegoose/typegoose/assets/10191085/8b23290e-0701-4beb-a086-63cafd11f2c2)
